pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,576 @@
+            "The Pareto/NBD model is still experimental. Please see code examples in documentation if model fitting issues are encountered.",
+            UserWarning,
+        )
+
+        if len(np.unique(customer_id)) != len(customer_id):
+            raise ValueError("Customers must have unique ID labels.")
+
+        super().__init__()
+
+        self._customer_id = customer_id
+        self._frequency = frequency
+        self._recency = recency
+        self._T = T
+
+        r_prior, alpha_prior, s_prior, beta_prior = self._process_priors(
+            r_prior, alpha_prior, s_prior, beta_prior
+        )
+        # TODO: rename hyperpriors to purchase_shape, purchase_scale, churn_shape, churn_scale?
+        coords = {"customer_id": self._customer_id}
+        with pm.Model(coords=coords) as self.model:
+            # purchase rate priors
+            r = self.model.register_rv(r_prior, name="r")
+            alpha = self.model.register_rv(alpha_prior, name="alpha")
+
+            # churn priors
+            s = self.model.register_rv(s_prior, name="s")
+            beta = self.model.register_rv(beta_prior, name="beta")
+
+            ParetoNBD(
+                name="likelihood",
+                r=r,
+                alpha=alpha,
+                s=s,
+                beta=beta,
+                T=self._T,
+                observed=np.stack((self._recency, self._frequency), axis=1),
+                dims="customer_id",
+            )
+
+    def _process_priors(self, r_prior, alpha_prior, s_prior, beta_prior):
+        # priors for purchase rate
+        if r_prior is None:
+            r_prior = pm.Weibull.dist(alpha=10, beta=1)
+        else:
+            self._check_prior_ndim(r_prior)
+        if alpha_prior is None:
+            alpha_prior = pm.Weibull.dist(alpha=10, beta=10)
+        else:
+            self._check_prior_ndim(alpha_prior)
+
+        # hyper priors for churn rate
+        if s_prior is None:
+            s_prior = pm.Weibull.dist(alpha=10, beta=1)
+        else:
+            self._check_prior_ndim(s_prior)
+        if beta_prior is None:
+            beta_prior = pm.Weibull.dist(alpha=10, beta=10)
+        else:
+            self._check_prior_ndim(beta_prior)
+
+        return super()._process_priors(r_prior, alpha_prior, s_prior, beta_prior)
+
+    def _unload_params(
+        self,
+    ) -> Tuple[Any, ...]:
+        """Utility function retrieving posterior parameters for predictive methods"""
+        return tuple([self.fit_result.posterior[param] for param in self._params])
+
+    # TODO: Convert to list comprehension to support covariates?
+    def _process_customers(
+        self,
+        customer_id: Union[np.ndarray, pd.Series, xarray.DataArray, None],
+        frequency: Union[np.ndarray, pd.Series, xarray.DataArray, None],
+        recency: Union[np.ndarray, pd.Series, xarray.DataArray, None],
+        T: Union[np.ndarray, pd.Series, xarray.DataArray, None],
+    ) -> Tuple[xarray.DataArray]:
```

Comment:
I think changing the return to `Tuple[xarray.DataArray, ...]` solves all the issues :) 

FYI: I tried to fix this but could not push it to the branch ... I hope I did not messed up things hehe 🙈